etl/steps/data/garden/health/2023-04-25/wgm_2018.py

```diff
@@ -0,0 +1,353 @@
+    df_countries = _make_df_with_share_answers(df)
+    # Obtain dataframe for World
+    log.info("wgm_mental_health: building dataframe for World")
+    df_world = df.assign(country="World")
+    df_world = _make_df_with_share_answers(df_world, "weight_inter_country")
+    # # Obtain dataframe for continents
+    log.info("wgm_mental_health: building dataframe for continents")
+    df_continents = _build_df_with_continents(df)
+    df_continents = _make_df_with_share_answers(df_continents, "weight_inter_country")
+    # # Obtain dataframe for income groups
+    log.info("wgm_mental_health: building dataframe for income groups")
+    df_incomes = _build_df_with_incomes(df)
+    df_incomes = _make_df_with_share_answers(df_incomes, "weight_inter_country")
+    # # Merge
+    df = pd.concat([df_countries, df_world, df_continents, df_incomes], ignore_index=True)
+    # df = df_countries
+    # df = df.astype({"gender": "category", "age_group": "category"})
+    return df
+
+
+def map_ids_to_labels(df: pd.DataFrame) -> pd.DataFrame:
+    """Instead of having IDs we use labels.
+
+    The source uses IDs for gender and age groups, answers and questions.
+    """
+    # Answer ID to Answer label mapping
+    # Create unique identifier for answer id. Note that answer ids mean different things depending on the question!
+    # Therefore, we build a mapping `questionId__answerId -> answerLabel`
+    question_answer_id_to_label = {}
+    for q_id, q_props in MAPPING_QUESTION_VALUES.items():
+        for a_id, a_label in q_props["answers"].items():
+            question_answer_id_to_label[f"{q_id}__{a_id}"] = a_label
+    df["question__answer"] = df["question"] + "__" + df["answer"]
+
+    log.info("wgm_2018: sanity checking IDs (question, answer, gender, age_group)}")
+    # Sanity checks (Question)
+    _sanity_check_question(df)
+    # Sanity checks for (question, answer) pairs
+    _sanity_check_answer(df)
+    # # Sanity check gender IDs
+    _sanity_check_gender(df)
+
+    # Question ID to Question label mapping
+    question_id_to_label = {k: v["title"] for k, v in MAPPING_QUESTION_VALUES.items()}
+
+    # Map IDs to Labels (Question, Answer, Gender, Age group)
+    log.info("wgm_2018: mapping ids to labels}")
+    df["question"] = df["question"].replace(question_id_to_label)
+    df["answer"] = df["question__answer"].map(question_answer_id_to_label).fillna("Unknown")
+    df["gender"] = df["gender"].replace(MAPPING_GENDER_VALUES)
+    return df
+
+
+def filter_rows_with_low_participation(df: pd.DataFrame) -> pd.DataFrame:
+    """Filter rows where the number of answers for a question from a demographic group was very low"""
+    log.info("wgm_mental_health: Filtering entries with few participants.")
+    col_idx = ["country", "year", "question", "gender", "age_group"]
+    df_count = df.groupby(col_idx, observed=True, as_index=False)[["count"]].sum()
+    df = df.merge(df_count, on=col_idx, suffixes=("", "_total"), how="left")
```

Comment:
If I understand correctly, you could instead do something like:
```
df["count_total"] = df.groupby(col_idx, observed=True, as_index=False)["count"]. transform("sum")
```
(I haven't checked)

Reply:
Thanks! If using with `as_index=True` (default value) it works!

```python
df["count_total"] = df.groupby(col_idx, observed=True)["count"]. transform("sum")
```